WcaOnRails/app/models/competition.rb

```diff
@@ -1219,28 +1237,39 @@ def set_wcif_events!(wcif_events, current_user)
 
   # Takes an array of partial Person WCIF and updates the fields that are not immutable.
   def update_persons_wcif!(wcif_persons, current_user)
+    registrations = self.registrations.includes [
+      { assignments: [:schedule_activity] },
+      :user,
+      :registration_competition_events,
+    ]
+    competition_activities = all_activities
     wcif_persons.each do |wcif_person|
-      registration = registrations.find_by(user_id: wcif_person["wcaUserId"])
+      registration = registrations.find { |reg| reg.user_id == wcif_person["wcaUserId"] }
```

Comment:
Is this change because we already have loaded all the registrations, and doing a `find_by` would cause us to go back to the database?

Reply:
Yeah, that's it! It's a bit of a shame that Rails doesn't check for pre-loaded data to behave the same, but I can imagine it being a problem in other cases.